### PR TITLE
Add more details in Log4j default settings

### PR DESCRIPTION
**Pull request by lucyge2022**
### What changes are proposed in this pull request?

Add filename:linenumber:threadid details into default log4j settings so logs could be more informative when doing debugging.

### Why are the changes needed?

During debugging, it is hard to trace events for one particular file/event/thread with current default log4j settings

### Does this PR introduce any user facing changes?

N/A

### CODE REVIEW

**Comment by yuzhu** [comment]
@lucyge2022 Do we know if this adds overhead to the logging process? i am assuming not much, because line number and file names are static and thread id is a threadlocal var call.

**Comment by lucyge2022** [comment]
> @lucyge2022 Do we know if this adds overhead to the logging process? i am assuming not much, because line number and file names are static and thread id is a threadlocal var call.

I think this is nothing compared to with problems caused by RollingFileAppender which internally have lock contention when dealing with large concurrency:
https://stackoverflow.com/questions/50747813/log4j-rollingfileappender-performance
AsyncAppender with lock-free settings is more recommended nowadays : https://logging.apache.org/log4j/2.x/manual/async.html
https://logging.apache.org/log4j/2.x/performance.html#logging-library-performance-comparison

If we haven't seen any log panicking similar to this, where all threads are stuck at log4j trying to compete for locks, this pattern change should be fine, and we gain more debugging benefits than overhead
https://issues.apache.org/jira/browse/LOG4J2-953

File: conf/log4j.properties [hunk 1]
```diff
@@ -31,7 +31,7 @@ log4j.appender.=org.apache.log4j.varia.NullAppender
 log4j.appender.Console=org.apache.log4j.ConsoleAppender
 log4j.appender.Console.Target=System.out
 log4j.appender.Console.layout=org.apache.log4j.PatternLayout
-log4j.appender.Console.layout.ConversionPattern=%d{ISO8601} %-5p %c{1} - %m%n
+log4j.appender.Console.layout.ConversionPattern=%d{ISO8601} %-5p %c{1}[%t](%F:%L) - %m%n
```
Comment (jiacheliu3) [inline conf/log4j.properties:34]: Does this make sense? `%c{1}[%t](%F:%L)` -> `%c{1}:%L[%t]`
I just mean the class name and filename may be duplicate

**Review by jiacheliu3 — APPROVED** [review]
Just a quick comment on the format. Up to your judgement and I'm fine with the explanation on perf :)
+1 to Async appender!

File: conf/log4j.properties (hunk 1 above)
Comment (lucyge2022) [reply conf/log4j.properties:34]: sure yeah that make sense, let me remove it

Comment (lucyge2022) [reply conf/log4j.properties]: logs will be like this now for reference: 
`2023-03-16 07:57:07,786 INFO  [main](AlluxioWorkerProcess.java:246) - Alluxio worker started. id=2561991045737941628, bindHost=0.0.0.0, connectHost=192.168.0.189, rpcPort=29999, webPort=30000`

**Review by yuzhu — APPROVED** [review]

**Comment by Xenorith** [comment]
alluxio-bot, merge this please

**Pull request merged**